File: opus/ByteWorks/ORCA/Toolbox.Pascal/Lesson.1/Prob.1.4.pas
program MousePosition (output);uses Common, QuickDrawII, EventMgr;var   done: boolean;                       {are we done, yet?}   myEvent: eventRecord;                {event record}   h, v: integer;                       {old position (to avoid flicker)}   procedure WriteMousePos;   { Write the position of the mouse                            }   var      r: rect;                          {rectangle for area to erase}   begin {WriteMousePos}   if (h <> myEvent.eventWhere.h) or (v <> myEvent.eventWhere.v) then begin      SetSolidPenPat(black);            {erase any old stuff}      r.left := 0;  r.right := 150;      r.top := 0;   r.bottom := 15;      PaintRect(r);      MoveTo(10, 10);                   {write the position}      h := myEvent.eventWhere.h;      v := myEvent.eventWhere.v;      write(h, ', ', v);      end; {if}   end; {WriteMousePos}beginStartDesk(640);PenNormal;InitCursor;done := false;h := -1;repeat   if GetNextEvent(everyEvent, myEvent) then      if myEvent.eventWhat = keyDownEvt then         done := true;   WriteMousePos;until done;EndDesk;end.

File: opus/ByteWorks/ORCA/Toolbox.C/Lesson.1/Prob.1.4.cc
/***************************************************************** Mouse Position** Show the current mouse position****************************************************************/#pragma lint -1#include <stdio.h>#include <orca.h>#include <Event.h>#include <QuickDraw.h>BOOLEAN done;                           /* are we done, yet? */EventRecord myEvent;                    /* event record */int h = -1, v;                          /* old position (to avoid flicker) *//***************************************************************** WriteMousePos** Write the position of the mouse.****************************************************************/void WriteMousePos (void){Rect r;                                 /* rectangle for area to erase */if ((h != myEvent.where.h) || (v != myEvent.where.v)) {   SetSolidPenPat(black);               /* erase any old stuff */   r.h1 = 0;  r.h2 = 150;   r.v1 = 0;  r.v2 = 15;   PaintRect(&r);   MoveTo(10, 10);                      /* write the position */   h = myEvent.where.h;   v = myEvent.where.v;   printf("%d, %d", h, v);   }}/***************************************************************** Main program****************************************************************/int main (void){startdesk(640);                         /* start the tools */PenNormal();                            /* set up normal pen parameters */InitCursor();                           /* start the arrow cursor */done = FALSE;                           /* we aren't done, yet */while (!done) {                                        /* handle the events */   if (GetNextEvent(everyEvent, &myEvent))      if (myEvent.what == keyDownEvt)         done = TRUE;   WriteMousePos();   }enddesk();}